### PR TITLE
Image controller

### PR DESCRIPTION
**Pull request by gbenhaim**
Add image-controller
    
    1. Add the manifests.
    2. Add a deploy script.
    3. Update the README.
    
    Since the image controller requires a Quay organization
    and application, to not add an additional prerequisites for trying out
    Konflux, the image-controller deployment isn't included in the "deploy-konflux.sh" script, but rather on its own script.

depends on https://github.com/konflux-ci/konflux-ci/pull/103

### CODE REVIEW

**Comment by gbenhaim** [comment]
@yftacherzog I need to generate the index locally?

**Comment by yftacherzog** [comment]
> @yftacherzog I need to generate the index locally?
> 

Yes. We had issues with having the automation push changes or the time being, so instead the automation verifies the table of contents is up to date

File: README.md [hunk 1]
```diff
@@ -387,6 +388,40 @@ E.g. [Docker Hub](https://hub.docker.com/), [Quay.io](https://quay.io/repository
...
+
+#### Automatically Provision Quay Repositories For Container Images
+
+**Note**: This step is mandatory for importing components using the UI.
+
+Konflux integrates with the [Image Controller](https://github.com/konflux-ci/image-controller)
+that can automatically create Quay repositories when onboarding a component.
+
+The controller requires a token for a Quay application that has the following permissions on an
+organization:
```
Comment (yftacherzog) [inline README.md]: This suggests that we're going to describe the permissions, but instead we provide a procedure.

File: README.md [hunk 2]
```diff
@@ -387,6 +388,40 @@ E.g. [Docker Hub](https://hub.docker.com/), [Quay.io](https://quay.io/repository
...
+
+1. [Create a user on Quay.io](https://quay.io/)
+
+2. [Create Quay Organization](https://docs.projectquay.io/use_quay.html#org-create)
+
+3. [Create Quay Application](https://docs.projectquay.io/use_quay.html#_create_oauth_access_token) - The application should have the following permissions
+
+   3.1 Administer Organization
+   3.2 Administer Repositories
+   3.3 Create Repositories
```
Comment (yftacherzog) [inline README.md]: Those are rendered on the same line.

File: README.md [hunk 3]
```diff
@@ -387,6 +388,40 @@ E.g. [Docker Hub](https://hub.docker.com/), [Quay.io](https://quay.io/repository
...
+
+4. Run the `deploy-image-controller.sh` script:
+
+`$TOKEN` - the token for the application you've created on step 3.
+
+`$Orgnization` - the name of the organization you created on step 2.
```
Comment (yftacherzog) [inline README.md]: There's a typo on the word, and it should be capitalized.
```suggestion
`$ORGANIZATION` - the name of the organization you created on step 2.
```

File: deploy-image-controller.sh [hunk 4]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash -e
+
+script_path="$(dirname -- "${BASH_SOURCE[0]}")" 
+
+main() {
+    local token="${1:?A token for quay should provided}"
+    local org=${2:?Quay orgnization name should be provided}
```
Comment (yftacherzog) [inline deploy-image-controller.sh]:
```suggestion
    local org=${2:?Quay organization name should be provided}
```

File: deploy-image-controller.sh [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    create_secret "$token" "$org"
+    deploy
+}
+
+create_secret() {
+    local token="${1:?A token for quay should provided}"
+    local org=${2:?Quay orgnization name should be provided}
```
Comment (yftacherzog) [inline deploy-image-controller.sh]:
```suggestion
    local org=${2:?Quay organization name should be provided}
```

File: README.md [hunk 6]
```diff
@@ -387,6 +388,40 @@ E.g. [Docker Hub](https://hub.docker.com/), [Quay.io](https://quay.io/repository
...
+
+```bash
+
+./deploy-image-controller.sh $TOKEN $ORGNIZATION
+```
```
Comment (yftacherzog) [inline README.md]:
```suggestion
```bash
./deploy-image-controller.sh $TOKEN $ORGNIZATION
```
```

File: README.md [hunk 7]
```diff
@@ -387,6 +388,40 @@ E.g. [Docker Hub](https://hub.docker.com/), [Quay.io](https://quay.io/repository
    finishes successfully, and that your public repository contains the images pushed by
    the pipeline.
 
+
+#### Automatically Provision Quay Repositories For Container Images
```
Comment (yftacherzog) [inline README.md]:
```suggestion
#### Automatically Provision Quay Repositories for Container Images
```

File: deploy-image-controller.sh [hunk 8]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash -e
+
+script_path="$(dirname -- "${BASH_SOURCE[0]}")" 
+
+main() {
+    local token="${1:?A token for quay should provided}"
```
Comment (yftacherzog) [inline deploy-image-controller.sh]:
```suggestion
    local token="${1:?A token for quay should be provided}"
```

File: deploy-image-controller.sh [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    local org=${2:?Quay orgnization name should be provided}
+
+    create_secret "$token" "$org"
+    deploy
+}
+
+create_secret() {
+    local token="${1:?A token for quay should provided}"
```
Comment (yftacherzog) [inline deploy-image-controller.sh]:
```suggestion
    local token="${1:?A token for quay should be provided}"
```

File: deploy-image-controller.sh [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    local org=${2:?Quay orgnization name should be provided}
+    local ns=image-controller
+    local secret=quaytoken
+
+    if kubectl get secret "$secret" -n "$ns" &> /dev/null; then
+        echo "Image controller quay secret already exist" >&2
```
Comment (yftacherzog) [inline deploy-image-controller.sh]:
```suggestion
        echo "Image controller quay secret already exists" >&2
```

File: konflux-ci/build-service/core/apps_v1_deployment_build-service-controller-manager.yaml [hunk 11]
```diff
@@ -27,7 +27,7 @@ spec:
         - --leader-elect
         command:
         - /manager
-        image: quay.io/redhat-appstudio/build-service:29b09e5d8350f75c92d5796fbbf62647fe28610b
+        image: quay.io/konflux-ci/build-service
```
Comment (yftacherzog) [inline konflux-ci/build-service/core/apps_v1_deployment_build-service-controller-manager.yaml:30]: Are we ok with not providing the tag?

File: README.md (hunk 1 above)
Comment (gbenhaim) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (gbenhaim) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (gbenhaim) [reply README.md]: donee

File: deploy-image-controller.sh (hunk 4 above)
Comment (gbenhaim) [reply deploy-image-controller.sh]: done

File: deploy-image-controller.sh (hunk 5 above)
Comment (gbenhaim) [reply deploy-image-controller.sh]: done

File: README.md (hunk 6 above)
Comment (gbenhaim) [reply README.md]: done

File: README.md (hunk 7 above)
Comment (gbenhaim) [reply README.md]: done

File: deploy-image-controller.sh (hunk 8 above)
Comment (gbenhaim) [reply deploy-image-controller.sh]: done

File: deploy-image-controller.sh (hunk 9 above)
Comment (gbenhaim) [reply deploy-image-controller.sh]: done

File: deploy-image-controller.sh (hunk 10 above)
Comment (gbenhaim) [reply deploy-image-controller.sh]: done

File: README.md [hunk 12]
```diff
@@ -387,6 +388,36 @@ E.g. [Docker Hub](https://hub.docker.com/), [Quay.io](https://quay.io/repository
...
 
+
+#### Automatically Provision Quay Repositories for Container Images
+
+**Note**: This step is mandatory for importing components using the UI.
+
+Konflux integrates with the [Image Controller](https://github.com/konflux-ci/image-controller)
+that can automatically create Quay repositories when onboarding a component.
+The image controller requires an access to a Quay organization.
+Please follow the following step for configuring it:
```
Comment (yftacherzog) [inline README.md]:
```suggestion
Please follow the following steps for configuring it:
```

Comment (gbenhaim) [reply README.md]: done

**Review by yftacherzog — APPROVED** [review]
lgtm

**Pull request merged**